plugins/jvm-example/src/main/java/org/elasticsearch/plugin/example/JvmExamplePlugin.java

```diff
@@ -111,5 +117,16 @@ public Settings additionalSettings() {
     public void onModule(RepositoriesModule repositoriesModule) {
     }
 
-
+    /**
+     * Module decalaring some example configuration and a _cat action that uses
+     * it.
+     */
+    public static class ConfiguredExampleModule extends AbstractModule {
+        @Override
+        protected void configure() {
+          bind(ExamplePluginConfiguration.class).asEagerSingleton();
+          Multibinder<AbstractCatAction> catActionMultibinder = Multibinder.newSetBinder(binder(), AbstractCatAction.class);
```

Comment:
> This is not a valid argument for not making the system better with well defined extension points. How can we get to a point where we do have well defined extension points, if we don't start somewhere?

I guess my argument is that Multibinder works regardless of how much work we put into supporting it but onModule only works if we make it work and we aren't doing much with it.

> How so? A Set, a method, and a couple lines in configure?

I'll push the commit that gets it working in a minute - its just a few lists and methods to add to them.  Compared to the 2 lines in the plugin. I guess the hard part for the plugin author is to onModule the _right_ module. I had to do more debugging Elasticsearch to get the onModule thing working then I did the Multibinder - Guice just worked.

I just don't get why we wrap Guice here - its built for this kind of thing.